pkg/machine/e2e/machine_test.go

```diff
@@ -55,13 +57,23 @@ var _ = BeforeSuite(func() {
 	if err != nil {
 		Fail("unable to create new download")
 	}
+
 	if len(downloadLocation) < 1 {
-		fcd, err := dd.GetFCOSDownload(defaultStream)
-		if err != nil {
-			Fail("unable to get virtual machine image")
+		if provider.VMType() != machine.WSLVirt {
+			fcd, err := dd.GetFCOSDownload(defaultStream)
+			if err != nil {
+				Fail("unable to get virtual machine image")
+			}
+			downloadLocation = fcd.Location
+		} else {
+			fd, err := wsl.NewFedoraDownloader(provider.VMType(), "", defaultStream.String())
+			if err != nil {
+				Fail("unable to get WSL virtual image")
+			}
+			downloadLocation = fd.Get().URL.String()
```

Comment:
this needs to put put behind the windows build tag as the wsl backend can only be compiled on windows.